### ISSUE TITLE
Expand logging within app

### ISSUE THREAD

**Issue by ghachey** [PacificEMIS/pacific-emis-education-survey-tool#40]
Currently only logging Delete surveys. Add logging for the following operations:

- Creating new surveys
- Merging survey into another tablet (tag it "Merged To")
- Merging survey from another tablet (tag it "Merged From")
- Change Date of a Survey

Add ability to click on log for more details in a pop-up.

**Comment by maratomega** [comment]
Hi @ghachey please clarify, what info we should display in pop-up.
These mentioned operations should be dispayed separately in Logs, correct:

- Creating new surveys

- Merging survey into another tablet (tag it "Merged To")

- Merging survey from another tablet (tag it "Merged From")

- Change Date of a Survey

**Comment by ghachey** [comment]
> Hi @ghachey please clarify, what info we should display in pop-up. These mentioned operations should be dispayed separately in Logs, correct:
> 
>     * Creating new surveys
> 
>     * Merging survey into another tablet (tag it "Merged To")
> 
>     * Merging survey from another tablet (tag it "Merged From")
> 
>     * Change Date of a Survey

No, the actual logs would be summary only showing the date time and the logs description. Clicking on it could provide more details such as Date time, Log description, Tablet ID, User, Anything else? This is what is meant by additional information. Additional information for an individual logs unless you can display all the relevant log details into a row showing without truncating text, etc. then this display of additional information would become unnecessary.

**Comment by ghachey** [comment]
@maratomega please share progress on this one.

**Comment by ghachey** [comment]
Some notes:

- Currently the Logs "table" display the columns "ID/Date", "School", "Email", "Type", "Status". Let's simplify this to only "Date/Time" (i.e. the logged operation), "School", "Period" (I am adding this one as it is important to identify the actual survey in question), "Status" (but status should be renamed to "Description". I don't like the term status for a log entry.) and finally "Type" (e.g. SCHOOL_ACCREDITATION, WASH, etc.)
- User can click on the log entry to display more details. This should include "Tablet ID", "Date/Time" (of logged operation), "School ID", "School", "Period", "Description" (the log description like 'Survey deleted' , 'Survey created', etc.), "User" (this actually what you called email but should be User and be based on Created User or Last Edited User which ever is the relevant one), "Type" (e.g. SCHOOL_ACCREDITATION, WASH, etc.)

Open question: where does the log data get stored? Just locally on tablet? That would be fine for now. Eventually it could be useful to also push to the centralize DB in a simple audit table through a POST REST API endpoint.

**Comment by maratomega** [comment]
@ghachey in design we adjusted Logs screen columns' names and would like to get your approval on the way how we are going to display the additional info 
![Screenshot 2023-12-15 at 17 14 22](https://github.com/PacificEMIS/pacific-emis-education-survey-tool/assets/108275362/b344e447-8a51-4e58-8fcc-b54c7349c618)

**Comment by ghachey** [comment]
@maratomega That looks great. Couple of details you are probably aware of already:

- I'm not sure it is possible to have two periods. I thought the Period was a single date.
- The Tablet ID would not be the school ID but the actual tablet ID. We worked on adding this unique tablet identifier in a previous work recently. This will be the one there.

**Comment by maratomega** [comment]
@ghachey Can you please clarify, about "Period", we assume that "Period" stands for the time frame when Survey was created (create date) and when Survey was edited/changed/deleted (edit date), that's why in the screenshot we showed 2 dates (one date duplicates the date from the column "Date") in the "Period" column.
Correct me if I'm wrong please, in the "Period" column you want to show only the date when Survey was edited/changed/deleted (edit date)?

**Comment by ghachey** [comment]
> @ghachey Can you please clarify, about "Period", we assume that "Period" stands for the time frame when Survey was created (create date) and when Survey was edited/changed/deleted (edit date), that's why in the screenshot we showed 2 dates (one date duplicates the date from the column "Date") in the "Period" column. Correct me if I'm wrong please, in the "Period" column you want to show only the date when Survey was edited/changed/deleted (edit date)?

I am not clear on what you mean. The period is a single date that is used more like to identify the survey. In the metadata you have called this the surveyTag. So for Period you use the surveyTag (highlighted in yellow below). For the Date/Time just the date time of the logged operation so this we could use the lastEditedDateTime (highlighted in orange below) from the metadata, no?

![image](https://github.com/PacificEMIS/pacific-emis-education-survey-tool/assets/1091769/7d7dcce9-bca9-4159-bf19-82ce96f12744)

**Comment by maratomega** [comment]
@ghachey Now its clear, thanks for the clarification.
But maybe we need to rename the column "Period" to something like "Created" name?

**Comment by ghachey** [comment]
@maratomega The new logging is good but I do not see a log for a Remove of a Survey which funny enough is the only log we had before. Now it logs more but no longer logs removing a survey. Can you re-add this to the things that get logged please.

**Comment by ghachey** [comment]
@maratomega When you "Remove" a survey from the app no logging takes place. In other words, the below does not get logged. Note this is necessary. And it would be really useful to log any "Change date" as well if it is quick.

![Screenshot 2024-05-07 183108](https://github.com/PacificEMIS/pacific-emis-education-survey-tool/assets/1091769/dc2710fd-8246-4bd3-9682-8a667dd85a1a)

**Comment by maratomega** [comment]
@ghachey please clarify, if you want to add another log - Change time. We need to create a different section(item) for it?
Date/time and Period are different items, correct?

![567575756](https://github.com/PacificEMIS/pacific-emis-education-survey-tool/assets/108275362/26643330-46f5-421d-b227-7f74f4f52adc)

**Comment by ghachey** [comment]
> @ghachey please clarify, if you want to add another log - Change time. We need to create a different section(item) for it? Date/time and Period are different items, correct?
> 
> ![567575756](https://private-user-images.githubusercontent.com/108275362/328523015-26643330-46f5-421d-b227-7f74f4f52adc.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUxMzU5MzgsIm5iZiI6MTcxNTEzNTYzOCwicGF0aCI6Ii8xMDgyNzUzNjIvMzI4NTIzMDE1LTI2NjQzMzMwLTQ2ZjUtNDIxZC1iMjI3LTdmNzRmNGY1MmFkYy5QTkc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUwOFQwMjMzNThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMjM0YTc3NGFiMTM0YTFhMjIxNzU3OGNlOTc4MzU5ZTI5NTdkMzZmZDFiMjJjNTk5NWM2MjlhMzQ0NTBlOGJhJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.XUdLVsV4GjxzcVBuk8F8QrvyCE-XWNbIFbu-G4yLKbg)

This is mentioned a few places in this issue and elsewhere. I think it now warrants its own special place in the official developer documentation and this is where I will refer you to [
https://docs.pacific-emis.org/doku.php?id=education_survey_tool_android_app_developers_manual#period_vs_create_and_edit_timestamps](url)

It would be great to log this operation when it happens.

**Comment by ghachey** [comment]
Logging feature adequate for now.
